### PR TITLE
Refactor apply gate

### PR DESCRIPTION
**Pull request by SamDuffield**
Remove `_get_apply_gate` - I think it's much cleaner now

Users can now also manually do `new_statetensor = apply_gate(old_statentensor, gate_unitary, qubit_inds)` if they want

### CODE REVIEW

File: qujax/version.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '0.2.8'
+__version__ = '0.2.9'
```
Comment (cqc-melf) [inline qujax/version.py:1]: Is this still compatible with the old API? Don't looks like that to me?

Comment (SamDuffield) [reply qujax/version.py:1]: all previous `qujax` code should still work

we just have access to a new `qujax.apply_gate` function

File: qujax/circuit.py [hunk 2]
```diff
@@ -20,41 +19,73 @@ def __call__(self, statetensor_in: jnp.ndarray = None) -> jnp.ndarray:
...
+                                            Callable[[], jnp.ndarray]]])\
+        -> Sequence[Callable[[jnp.ndarray], jnp.ndarray]]:
     """
+    Ensures all gate_seq elements are functions that map (possibly empty) parameters
+    to a unitary tensor.
 
-    def apply_gate(statetensor: jnp.ndarray, params: jnp.ndarray) -> jnp.ndarray:
-        """
-        Applies {} gate to statetensor.
+    Args:
```
[the remark this answers is not in the document]
Comment (SamDuffield) [reply qujax/circuit.py:52]: Not sure which notation you are referring to?

[the remark this answers is not in the document]
Comment (cqc-melf) [reply qujax/circuit.py:52]: `Args:` instead of listing the params with :param ... :

[the remark this answers is not in the document]
Comment (SamDuffield) [reply qujax/circuit.py:52]: yep I belive this is the docstring notation we use throughout `qujax`
but that does remind me I need to add  `qujax.apply_gate` to the docs!

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**